lib/Service/ParticipantService.php

```diff
@@ -961,10 +961,10 @@ public function changeInCall(Room $room, Participant $participant, int $flags, b
 			$flags &= ~Participant::FLAG_WITH_VIDEO;
 		}
 
-		if ($flags !== Participant::FLAG_DISCONNECTED) {
+		if ($flags !== Participant::FLAG_DISCONNECTED && !$silent) {
 			$event = new ModifyParticipantEvent($room, $participant, 'inCall', $flags, $session->getInCall());
 			$this->dispatcher->dispatch(Room::EVENT_BEFORE_SESSION_JOIN_CALL, $event);
-		} else {
+		} elseif (!$silent) {
```

Comment:
end call can never be silent, so this is not needed
```suggestion
		} else {
```